core/src/main/java/org/fao/geonet/kernel/security/keycloak/KeycloakUtil.java

```diff
@@ -0,0 +1,40 @@
+package org.fao.geonet.kernel.security.keycloak;
+
+import org.apache.commons.lang.StringUtils;
+import org.fao.geonet.utils.Log;
+import org.springframework.beans.BeansException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.web.authentication.LoginUrlAuthenticationEntryPoint;
+
+import javax.annotation.PostConstruct;
+
+public class KeycloakUtil {
+    public static String signinPath = null;
+    private static LoginUrlAuthenticationEntryPoint loginUrlAuthenticationEntryPoint;
+
+    @Autowired
+    private LoginUrlAuthenticationEntryPoint loginUrlAuthenticationEntryPoint0;
+
+    @PostConstruct
+    private void init () {
+        loginUrlAuthenticationEntryPoint = this.loginUrlAuthenticationEntryPoint0;
+    }
+
+    public static String getSigninPath() {
+        if (signinPath == null) {
```

Comment:
It is used for caching.

`public static String signinPath = null;
`

It checks to see if signinPath still contains the initial value and if so then sets the values based on configuration values from [config-security.properties](https://github.com/geonetwork/core-geonetwork/blob/master/web/src/main/webapp/WEB-INF/config-security/config-security.properties#L27).
`loginForm=/signin?node=@@nodeId@@`

Subsequent calls should just return the value.

The intent was to use existing configuration values instead of hardcoding the value to "/signin"  But I did noticed that there are some hard coded values of "/signin" elsewhere in the code(mostly in Angular) so I'm not sure how easy it would be to change this property anyway